### ISSUE TITLE
COINBASE fetchBalance

### ISSUE THREAD

**Issue by BHruska** [ccxt/ccxt#20044]
### Operating System

Windows 11

### Programming Languages

Python

### CCXT Version

CCXT Version: 4.1.52

### Description

Another problem switching over to Coinbase Advanced from Coinbase Pro.  

The fetchBalance() function does not return any values for USD, or USDC, or USDT.  Therefore, I cannot tell if there's funds to trade.

This works with my other exchanges, as it did with Coinbase Pro.

### Code

```
  

```

**Comment by respectabiggle** [comment]
Check your API key, make sure it has 'read' for your USD / USDC / USDT wallets?

FWIW, fetchBalance() is working normally for me, tested just now (Javascript).

**Comment by BHruska** [comment]
Good idea, but that doesn't appear to be it.  My API key has USD, USDC selected.  I've got both in my account, but they don't appear in fetchBalances().  I'm using python.

![image](https://github.com/ccxt/ccxt/assets/3506963/d463f757-4a33-453b-a5ff-cc5d8d0c7e8b)

**Comment by BHruska** [comment]
Any ideas on this?  

It clearly isn't returning the USD balance on Coinbase using the python implementation.

**Comment by sc0Vu** [comment]
Hi @BHruska 

That's weird, it works for me. Can you call this function `fetch_accounts_v3` and see the USD balance?

**Comment by BHruska** [comment]
Seems to be the same result with fetch_accounts_v3.  I get every other wallet returned, but nothing for USD or USDC.

**Comment by legitalk** [comment]
Same for me, also after switching from Coinbase Pro to Advanced fetchBalance is not couting any open orders to the balance (used section).

**Comment by BHruska** [comment]
Just to confirm, we are talking about Coinbase Advanced here.  Coinbase Pro always worked fine.  But they have now switched every account over the Advanced, so this is important.

**Comment by BHruska** [comment]
Maybe it has to do with the LIMIT, as noted in this post.  Perhaps the USD account is greater than the limit returned, and somehow ccxt is thus leaving it out of the return set.  

https://forums.coinbasecloud.dev/t/gettin-usd-balance-in-accounts/1844

**Comment by BHruska** [comment]
That's it.  

I created a new API key with only BTC, USD selected and I get back the USD balance.

Can this be fixed?

**Comment by legitalk** [comment]
@BHruska for me working fine with v3.

**Comment by BHruska** [comment]
No it isn't.  Not if you have all assets selected in the api setup.  I had to implement this kind of a work around.

'''        if current_exchange == "COINBASE":
            MAX_LIMIT = 250
            balances = []
            balances = exchange.fetch_balance({"limit": MAX_LIMIT, "v3": True})
            cursor = balances["info"]["cursor"]
            while True:
                tmp = exchange.fetch_balance(
                    {"limit": MAX_LIMIT, "v3": True, "cursor": cursor}
                )
                balances.update(tmp)
                if not balances["info"]["has_next"]:
                    break
                cursor = balances["info"]["cursor"]
            return balances
        else:
            balances = exchange.fetchBalance()
            return balances

'''

**Comment by jtashiro** [comment]
For me, I replaced references to hardcoded magic number 100 with 250  in coinbase.py in the fetch_accounts* methods.  Is there a permanent fix for this in the works?

**Comment by luke-tt** [comment]
any update on this? Coinbase gotta be the worst around!
